### ISSUE TITLE
Implement automated weight window generation

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#1961]
Now that we have the capability to _use_ mesh-based weight windows, for convenience we also need a method to _generate_ them. @pshriwise has started some preliminary work for WW generation and we have an upcoming small project to bring this to completion.

**Comment by ojschumann** [comment]
Do you consider creating the weight windows by  a deterministic adjoint calculation? (CADIS / FW-CADIS)

With MCNP I had the impression, that weight windows from [advantg](https://rsicc.ornl.gov/codes/ccc/ccc8/ccc-854.html) are much easier and straightforward to use that the MC generated ones (I was told to iterate the WW in my course). I currently have no idea about the license of advantg, but it might be an idea to check, weather it can be coupled to OpenMC.

**Comment by makeclean** [comment]
Yes, brining on ADVANTG or similar would be beneficial, however it does come with complications. the 'seamless' part of ADVANTG is a package called LAVA which is the MCNP interoperability layer, so a similar capability would need to be linked in. However, depending on ADVANTG is tricky, its an export controlled code not openly available to all. For deep shielding, some 'detuning' of weight windows is necessary for performance reasons since 'perfect' weight windows naturally lead to long histories. 

Conceptually, iterative generation is simpler, and actually neater in terms of code, doesn't require any extra dependencies. An open equivalent to ADVANTG, I personally would welcome, I mean we have quite nice with OpenMOC, it would be nice to see what could be leveraged there?

**Comment by akolsek** [comment]
@ojschumann - my solution was to implement [global variance reduction](https://www.sciencedirect.com/science/article/pii/S0306454911003203) in 4 lines and a printing option that creates the weight windows file. In the first run you create the WW file, while in the second you overwrite it with new information and you can continue until all the voxels have acceptable statistical error.
